### PR TITLE
DOC-54404-GitOps Agent Proxy Support

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
Added a section Proxy Support for GitOps Agent

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by shivagowda — DISMISSED** [review]
lgtm

File: docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md [hunk 1]
```diff
@@ -80,7 +75,7 @@ Click **GitOps** **Agents**.
 
 Click **New GitOps Agent**. The Agent wizard appears.
 
-## Option: Harness GitOps Agent with No Existing Argo CD Project
+## Harness GitOps Agent with no existing Argo CD project
```
Comment (brian-f-harness) [inline docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md:78]:
```suggestion
## Harness GitOps Agent without an existing Argo CD project
```

File: docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md [hunk 2]
```diff
@@ -233,6 +228,98 @@ Also, whenever new entities are created in mapped Argo CD projects, they are add
 
 For steps on setting up the mapping and import, go to [Map Argo projects to Harness GitOps Projects](multiple-argo-to-single-harness.md).
 
+## Proxy support
+
+The GitOps Agent can work on environments where traffic is routed through a proxy. Perform the following steps to configure proxy support for GitOps Agent.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md:233]:
```suggestion
The Harness GitOps Agent can work on environments where traffic is routed through a proxy. Perform the following steps to configure proxy support for the agent.
```

File: docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md [hunk 3]
```diff
@@ -233,6 +228,98 @@ Also, whenever new entities are created in mapped Argo CD projects, they are add
...
+
+1. Make sure that the Agent is running in HTTP mode.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md:235]:
```suggestion
1. Make sure that the agent is running in HTTP mode.  
```

File: docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md [hunk 4]
```diff
@@ -233,6 +228,98 @@ Also, whenever new entities are created in mapped Argo CD projects, they are add
 
 For steps on setting up the mapping and import, go to [Map Argo projects to Harness GitOps Projects](multiple-argo-to-single-harness.md).
 
+## Proxy support
+
+The GitOps Agent can work on environments where traffic is routed through a proxy. Perform the following steps to configure proxy support for GitOps Agent.
+
+1. Make sure that the Agent is running in HTTP mode.  
+   To verify, check if the property/config `GITOPS_SERVICE_PROTOCOL` value is set to `HTTP1` in the `configmap({agentname}-agent)` present in the YAML you get after creating the Agent.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md:236]:
```suggestion
   To verify, check if the property/config `GITOPS_SERVICE_PROTOCOL` value is set to `HTTP1` in the `configmap({agentname}-agent)` present in the YAML after you create the agent.  
```

File: docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md [hunk 5]
```diff
@@ -233,6 +228,98 @@ Also, whenever new entities are created in mapped Argo CD projects, they are add
...
 For steps on setting up the mapping and import, go to [Map Argo projects to Harness GitOps Projects](multiple-argo-to-single-harness.md).
 
+## Proxy support
+
+The GitOps Agent can work on environments where traffic is routed through a proxy. Perform the following steps to configure proxy support for GitOps Agent.
+
+1. Make sure that the Agent is running in HTTP mode.  
+   To verify, check if the property/config `GITOPS_SERVICE_PROTOCOL` value is set to `HTTP1` in the `configmap({agentname}-agent)` present in the YAML you get after creating the Agent.  
+   `GITOPS_SERVICE_PROTOCOL: HTTP1`
+2. Add a property/config `HTTPS_PROXY` and add proxy details such as URL, port, and auth details as its value in the configmap mentioned in Step 1. For example, `HTTPS_PROXY: "https://squid.proxy-test:3128"`.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md:238]:
```suggestion
2. Add a property/config `HTTPS_PROXY`, and add proxy details, such as URL, port, and auth details as its value in the configmap mentioned in Step 1. For example, `HTTPS_PROXY: "https://squid.proxy-test:3128"`.
```

File: docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md [hunk 6]
```diff
@@ -233,6 +228,98 @@ Also, whenever new entities are created in mapped Argo CD projects, they are add
...
+3. Add an environment variable `NO_PROXY` in the GitOps Agent deployment with the following value.
```
Comment (brian-f-harness) [inline docs/continuous-delivery/gitops/install-a-harness-git-ops-agent.md:239]:
```suggestion
3. Add an environment variable `NO_PROXY` in the Harness GitOps Agent deployment with the following value.  
```

**Review by brian-f-harness — APPROVED** [review]
lgtm. Left a couple minor suggestions.

**Pull request merged**